ftplugin/go/splitjoin.vim

```diff
@@ -11,5 +11,4 @@ let b:splitjoin_join_callbacks = [
       \ 'sj#go#JoinVars',
       \ 'sj#go#JoinStruct',
       \ 'sj#go#JoinFunc',
-      \ 'sj#go#JoinFuncCall',
```

Comment:
minor: I'm guessing, `sj#go#JoinFunc` joins both func definitions and func calls?
Since we have 2 separate functions for splitting (namely `sj#go#SplitFunc` and `sj#go#SplitFuncCall`), it might be a good idea to leave a comment somewhere describing why we don't have separate functions for joining.